validator/db/kv/attester_protection.go

```diff
@@ -115,3 +119,11 @@ func (store *Store) ApplyAttestationForPubKey(
 		return sourceEpochsBucket.Put(sourceEpochBytes, targetEpochBytes)
 	})
 }
+
+func isSurrounding(prevSource, prevTarget, incomingSource, incomingTarget uint64) bool {
+	return incomingSource < prevSource && incomingTarget > prevTarget
+}
+
+func isSurrounded(prevSource, prevTarget, incomingSource, incomingTarget uint64) bool {
+	return prevSource < incomingSource && prevTarget > incomingTarget
+}
```

Comment:
Some godocs would be helpful here. Perhaps linking to the definition of surround votes. 

According to [Proto's definition](https://github.com/protolambda/eth2-surround#definition)
```
s: source
t: target

a surrounds b if: s_a < s_b < t_b < t_a

s < t is a pre, so the condition can also be:

s_a < s_b and t_b < t_a
```

It would also help if the boolean logic order matched the definition so we could review this more easily.